src/interactive/Toaster.jsx

```diff
@@ -0,0 +1,136 @@
+import CSSTransitionGroup from 'react-addons-css-transition-group';
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+export const DELAY_TIME = 3000;
+
+/**
+ * @module Toaster
+ */
+class Toaster extends React.PureComponent {
+	constructor(props) {
+		super(props);
+
+		this.timeouts = [];
+		this.dismissedToasts = [];
+
+		this.cloneToast = this.cloneToast.bind(this);
+		this.setDismissedToast = this.setDismissedToast.bind(this);
+		this.clearTimeouts = this.clearTimeouts.bind(this);
+		this.setTimer = this.setTimer.bind(this);
+
+		this.state = {
+			toasts: this.props.toasts.map(this.cloneToast)
+		};
+	}
+
+	/**
+	 * @param {Object} dismissedToast - `Toast` components to remove from state
+	 * @returns undefined
+	 */
+	setDismissedToast(dismissedToast) {
+		this.setState({
+			toasts: this.state.toasts.filter(toast => dismissedToast.props.id !== toast.props.id)
+		});
+	}
+
+	/**
+	 * @returns undefined
+	 *
+	 * sets timers to delay each toast's dismissal
+	 */
+	setTimer() {
+		const toastsToDismiss = this.state.toasts.filter((toast) => toast.props.autodismiss);
+
+		toastsToDismiss.forEach((toast, i) => {
+			this.timeouts.push(setTimeout(() => {
+				this.setDismissedToast(toast);
+			}, DELAY_TIME*(i+1)));
```

Comment:
just throwing it out there - instead of multiplying the delay time, what do you think of having a shorter delay for each subsequent toast?, e.g. instead of the timeline for 3 toasts looking like

```
-------|
---------------|
-----------------------|
```

it would look like

```
-------|
----------|
-------------|
```

the equation would be something like

```js
const BASE_DELAY = 3000;
const MARGINAL_DELAY = 1000; // each additional toast will wait this much longer than the previous one
setTimeout(() => { ... }, BASE_DELAY + (MARGINAL_DELAY * i))
```